app/models/course/lesson_plan/item.rb

```diff
@@ -40,6 +40,17 @@ def copy_attributes(other, time_shift)
     self.end_at = other.end_at + time_shift if other.end_at
   end
 
+  # Test if the lesson plan item has started for self directed learning.
+  #
+  # @return [Boolean]
+  def self_directed_started?
+    if course && course.self_directed_learning_max_time
+      start_at && start_at - course.self_directed_learning_max_time < Time.zone.now
```

Comment:
Since we have database `null:false`, normally it won't. When you assign it to `nil` and call this method, it will break, here is just for safe.